photon-api/src/main/scala/com/linkedin/photon/ml/optimization/game/GameOptimizationConfigSnapshot.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.optimization.game
+
+/**
+  * Represents the complete Game optimization configuration as it appeared for a single run with a particular setting
+  * for each hyperparameter.
+  *
+  * @param fixedEffectOptimizationConfiguration the optimization configuration for each fixed effect
+  * @param randomEffectOptimizationConfiguration the optimization configuration for each random effect
+  * @param factoredRandomEffectOptimizationConfiguration the optimization configuration for each factored random effect
+  */
+case class GameOptimizationConfigSnapshot(
```

Comment:
We can't pass the full configuration object here, because we need the part that's specific to the current model (i.e. the specific regularization parameter values).